src/mem/metaslab.h

```diff
@@ -312,7 +323,32 @@ namespace snmalloc
       // https://github.com/CTSRD-CHERI/llvm-project/issues/588
       return sizeclass_t::from_raw(
         static_cast<size_t>(remote_and_sizeclass) &
-        (alignof(RemoteAllocator) - 1));
+        (REMOTE_WITH_BACKEND_MARKER_ALIGN - 1));
+    }
+
+    MetaEntry(const MetaEntry&) = delete;
+
+    MetaEntry& operator=(const MetaEntry& other)
+    {
+      // Don't overwrite the boundary bit with the other's
+      meta = (other.meta & ~BOUNDARY_BIT) | (meta & BOUNDARY_BIT);
```

Comment:
Another ambiguous provenance case.

```suggestion
      meta = (other.meta & ~BOUNDARY_BIT) | static_cast<address_t>(meta & BOUNDARY_BIT);
```